staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -104,6 +104,19 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 	return ev, nil
 }
 
+// UpdateEventAttribs updates authorizer attributes according to attribs
+func UpdateEventAttribs(ev *auditinternal.Event, attribs authorizer.Attributes) {
+	if user := attribs.GetUser(); user != nil {
+		ev.User.Username = user.GetName()
```

Comment:
I'm confused about why we need a 3rd user info on the event -- what is the difference between `EffectiveUser` and `ImpersonatedUser`?